api/files/urls.py

```diff
@@ -6,4 +6,5 @@
     url(r'^(?P<file_id>\w+)/$', views.FileDetail.as_view(), name=views.FileDetail.view_name),
     url(r'^(?P<file_id>\w+)/versions/$', views.FileVersionsList.as_view(), name=views.FileVersionsList.view_name),
     url(r'^(?P<file_id>\w+)/versions/(?P<version_id>\w+)/$', views.FileVersionDetail.as_view(), name=views.FileVersionDetail.view_name),
+    url(r'^(?P<node_id>\w+)/list/(?P<provider>\w+)(?P<path>/(?:.*/)?)', views.FileList.as_view(), name=views.FileList.view_name),
```

Comment:
Yes, sorry. The File List view should handle the bulk PATCH request, rather than a root file view.